public/app/plugins/datasource/loki/components/LokiQueryField.tsx

```diff
@@ -1,13 +1,20 @@
 import React, { FunctionComponent } from 'react';
 import { LokiQueryFieldForm, LokiQueryFieldFormProps } from './LokiQueryFieldForm';
 import { useLokiSyntax } from './useLokiSyntax';
+import { useLokiForceRefresh } from './useLokiForceRefresh';
 
-const LokiQueryField: FunctionComponent<LokiQueryFieldFormProps> = ({ datasource, ...otherProps }) => {
+const LokiQueryField: FunctionComponent<LokiQueryFieldFormProps> = ({
+  datasource,
+  datasourceStatus,
+  ...otherProps
+}) => {
   const { isSyntaxReady, setActiveOption, refreshLabels, ...syntaxProps } = useLokiSyntax(datasource.languageProvider);
+  useLokiForceRefresh(datasourceStatus, refreshLabels);
```

Comment:
I think this hook should be called on `useLokiSyntax` hook level as this hook is responsible for all things that Loki query field requires. I find `useLokiForceRefresh` hook not returning anything being even more magical than hooks already are ;)

You can easily accomplish that by extending `useLokiSyntax` signature to be 
```ts 
useLokiSyntax(languageProvider: LokiLanguageProvider, dataSourceStatus: DatasourceStatus)
```

Reply:
I'll fix that